src/Http/Deserializer/Calendar/CalendarForEventDataValidator.php

```diff
@@ -29,33 +30,42 @@ public function validate(array $data)
             $messages['permanent'] = 'Permanent events are not supported.';
         }
 
+        if (isset($data['status'])) {
+            try {
+                (new UpdateStatusValidator())->validate($data['status']);
+            } catch (DataValidationException $dataValidationException) {
+                $messages['status'] = $dataValidationException->getValidationMessages();
+            }
+        }
+
         $messages = array_merge(
             $messages,
             (new TimeSpanValidator())->validate($data)
         );
 
-        $timeSpans = $calendarJSONParser->getTimeSpans($data);
+        // When the time spans contain errors it makes no sense to validate the rest of the data.
+        if (empty($messages)) {
```

Comment:
Nit: An early return could avoid this big `if` body.

Reply:
I think that with an early return we have to copy the throw of the validation exception twice?

This part needs to be done twice I think. Once when the status or time spans introduce errors or once when the start or end date introduce errors.
```
        if (!empty($messages)) {
            $e = new DataValidationException();
            $e->setValidationMessages($messages);
            throw $e;
        }
```

Reply:
Yes, but you could move the creation of that and throwing to a private method and do something like this in both places:
```
$this->throwIfNotEmpty($messages);
```

Reply:
Indeed! The early return makes the code better readable in combination with the helper method.
I did put the helper method `throwIfNotEmpty` inside a trait.